2017-10-24  Oliver Nathaniel Goch  <oliver@lnxsrv09.seas.ucla.edu>

	* ~/coreutils-with-bug/lib/timespec.h (timespec_cmp):
	Used comparison operators instead of subtracting to avoid overflow.
	This fixed the problem with the very old times

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ coreutils-with-bug/lib/timespec.h	2017-10-24 17:14:41.556485000 -0700
@@ -45,8 +45,19 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec > b.tv_sec)
+    {
+      return 1;
+    }
+  else if(a.tv_sec < b.tv_sec)
+    {
+      return -1;
+    }
+  else
+    {
+      return 0;
+    }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
